onnxruntime/core/mlas/inc/mlas_qnbit.h

```diff
@@ -159,14 +163,26 @@ MlasSQNBitGemmPackQuantBDataSize(
 /**
  * @brief Packs the quantized B data in a format that the kernel expects.
  *
+ * If the function is called without QuantBScale and QuantBZeroPoint,
+ * it just packs QuantBData into PackedQuantBDataAndOrBlkSum.
+ *
+ * If the function is called with QuantBData, QuantBScale, and QuantBZeroPoint
+ * additional BlkSum (Scale * zeropoint) is computed and stored at the second part of PackedQuantBDataAndOrBlkSum.
+ *
+ * Because ORT OpKernel::PrePack is called for each input (in this case, QuantBData,
+ * QuantBScale, and QuantBZeroPoint) separately, this function may be called 3 times, first with QuantBData,
+ * and then QuantBScale and QuantBZeroPoint. The second time the function is called with QuantBScale,
```

Comment:
nit: "The second time the function is called with QuantBScale," is redundant.

Reply:
depending on implementation, the buffer may hold only packed B weights (the NEON case), or it hold additional BlkSum(in case of avx).

Reply:
updated the comment

Reply:
updated the comment